cypresse2e/cypress/integration/e2eTests/signIn/signIn.spec.js

```diff
@@ -0,0 +1,162 @@
+            var passwordField = signIn.getPasswordField();
+            passwordField.should('be.visible').and('have.attr', 'name', 'password').focus().blur();
+            var passwordRequired = signIn.getPasswordRequired();
+            passwordRequired.should('be.visible');
+            passwordRequired.contains('Password is required.');
+
+        });
+
+        it('Password entered not meeting 8 characters triggers error', () => {
+            var passwordField = signIn.getPasswordField();
+            passwordField.should('be.visible');
+            passwordField.type('test');
+            var passwordLengthRequired = signIn.getPasswordLengthRequired();
+            passwordLengthRequired.should('be.visible');
+            passwordLengthRequired.contains('Password must be at least 8 characters');
+
+        });
+
+        it('Password entered not meeting special characters triggers error', () => {
+            var passwordField = signIn.getPasswordField();
+            passwordField.should('be.visible');
+            passwordField.type('testtest');
+            var passwordCharacterRequired = signIn.getPasswordCharacterRequired();
+            passwordCharacterRequired.should('be.visible');
+            passwordCharacterRequired.contains('Password must contain at least 3 of these: a lower-case letter, an upper-case letter, a number, a special character (such as !@#$%^&*).');
+
+        });
+
+        it('Sign in button is disabled when required details are not entered', () => {
+            var signInButton = signIn.getSignInButton();
+            signInButton.should('be.visible').and('have.attr', 'aria-disabled', 'true');
+
+        });
+
+        it('Sign in button is enabled when required details are entered', () => {
+            var emailField = signIn.getEmailField();
+            emailField.should('be.visible').and('have.attr', 'name', 'email');
+            emailField.type('qa.test@gmail.com');
+            var passwordField = signIn.getPasswordField();
+            passwordField.should('be.visible').and('have.attr', 'name', 'password');
+            passwordField.type('Testing!');
+            var signInButton = signIn.getSignInButton();
+            signInButton.should('be.visible').and('have.attr', 'aria-disabled', 'false');
+
+        });
+
+        it('Login fail alert appears when incorrect email & password are entered and submitted by user', () => {
+            var emailField = signIn.getEmailField();
+            emailField.should('be.visible').and('have.attr', 'name', 'email');
+            emailField.type('qa.test@test.com');
+            var passwordField = signIn.getPasswordField();
+            passwordField.should('be.visible').and('have.attr', 'name', 'password');
+            passwordField.type('WrongPW!wpw3hi');
+            var signInButton = signIn.getSignInButton();
+            signInButton.should('be.visible').and('have.attr', 'aria-disabled', 'false').click();
+
```

Comment:
- Please check the error message displayed after user clicks on SignIn button after entering wrong userID/pwd.

Reply:
@jessica-lau 
Agree with Rajbir about adding a test for "_after entering wrong userID/pwd_".

Reply:
One more thing. Maybe it would be good to have a separate file with constants like email "_qa.test@test.com_"and use them across all tests? Or this is specific only for this test?

Reply:
@rabbi08 and @galeza I added a test for the error alert and added a file for constants, to avoid having to type the email multiple times. Thanks!

Reply:
@jessica-lau 
When we run “Invalid ID/Pwd” test case multiple time, it locks that email ID and the error also changes as below, and it FAILs the test case

"Login failed, reason: Maximum number of sign in attempts exceeded."

Hence to validate “Invalid ID/Pwd” test case with the error message as "Login failed, reason: Wrong email or password.” I guess following test steps might resolve this issue:

1. Enter wrong email/ID and assert "Login failed, reason: Wrong email or password.” The test case will PASS here.
2. Then, to reset the login attempts/test data, login using valid email/ID, before you exit this test case to avoid the account lock for the test data.

Reply:
@rabbi08 I added changes to the invalid email/password test to prevent the account from locking out by signing into it after entering invalid credentials. I also added a separate test to test the exceeded attempts message -  I used a separate dummy email for this.